src/main/java/barry/Parser.java

```diff
@@ -0,0 +1,73 @@
+package barry;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * The Parser class is responsible for interpreting user input and converting it into executable commands.
+ * It parses different types of user commands and provides the appropriate Command object to be executed.
+ */
+public class Parser {
+
+    /**
+     * Parses the user input and returns the corresponding Command object.
+     *
+     * @param input The user input string to parse.
+     * @return The Command object that corresponds to the user's input.
+     * @throws BarryException If the input does not correspond to any known command, or if there is an error in the input format.
+     */
+    public static Command parse(String input) throws BarryException {
+        String[] parts = input.split(" ", 2);
+        String commandWord = parts[0];
+        String arguments = parts.length > 1 ? parts[1] : "";
+
+        switch (commandWord) {
+            case "bye":
+                return new Command.ExitCommand();
+            case "list":
+                return new Command.ListCommand();
+            case "mark":
+                return new Command.MarkCommand(Integer.parseInt(arguments));
+            case "unmark":
+                return new Command.UnmarkCommand(Integer.parseInt(arguments));
+            case "todo":
+                return new Command.AddTodoCommand(arguments);
+            case "find":
+                return new Command.FindCommand(arguments);
+            case "deadline":
+                String[] deadlineParts = arguments.split(" /by ");
+                if (deadlineParts.length < 2) throw new BarryException("The deadline format is incorrect.");
```

Comment:
Exception is well handled. However, for the coding standard we are following in this course I believe we are supposed to put the conditional statement in another line. For example:

```java
if (condition) {
    action;
}